ibm/service/power/resource_ibm_pi_network_security_group_rule.go

```diff
@@ -0,0 +1,654 @@
+		}
+		_, err = isWaitForIBMPINetworkSecurityGroupRuleAdd(ctx, nsgClient, nsgID, ruleID, d.Timeout(schema.TimeoutCreate))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		d.SetId(fmt.Sprintf("%s/%s/%s", cloudInstanceID, nsgID, ruleID))
+	}
+
+	return resourceIBMPINetworkSecurityGroupRuleRead(ctx, d, meta)
+}
+
+func resourceIBMPINetworkSecurityGroupRuleRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	cloudInstanceID, nsgID, err := splitID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	nsgClient := instance.NewIBMIPINetworkSecurityGroupClient(ctx, sess, cloudInstanceID)
+	networkSecurityGroup, err := nsgClient.Get(nsgID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.Set(Attr_Name, networkSecurityGroup.Name)
+	d.Set(Attr_CRN, networkSecurityGroup.Crn)
+
+	if len(networkSecurityGroup.Members) > 0 {
+		members := []map[string]interface{}{}
+		for _, mbr := range networkSecurityGroup.Members {
+			mbrMap := networkSecurityGroupMemberToMap(mbr)
+			members = append(members, mbrMap)
+		}
+		d.Set(Attr_Members, members)
+	} else {
+		d.Set(Attr_Members, nil)
+	}
+
+	d.Set(Attr_NetworkSecurityGroupID, networkSecurityGroup.ID)
+	if len(networkSecurityGroup.Rules) > 0 {
+		rules := []map[string]interface{}{}
+		for _, rule := range networkSecurityGroup.Rules {
+			ruleMap := networkSecurityGroupRuleToMap(rule)
+			rules = append(rules, ruleMap)
+		}
+		d.Set(Attr_Rules, rules)
+	}
+	if len(Attr_UserTags) > 0 {
+		d.Set(Attr_UserTags, networkSecurityGroup.UserTags)
+	}
+	return nil
+}
+
+func resourceIBMPINetworkSecurityGroupRuleDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	ids, err := flex.IdParts(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if len(ids) == 3 {
+		cloudInstanceID := ids[0]
+		nsgID := ids[1]
+		ruleID := ids[2]
+
+		sess, err := meta.(conns.ClientSession).IBMPISession()
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		nsgClient := instance.NewIBMIPINetworkSecurityGroupClient(ctx, sess, cloudInstanceID)
+
+		err = nsgClient.DeleteRule(nsgID, ruleID)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		_, err = isWaitForIBMPINetworkSecurityGroupRuleRemove(ctx, nsgClient, nsgID, ruleID, d.Timeout(schema.TimeoutDelete))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+	d.SetId("")
+	return nil
+}
+
+func isWaitForIBMPINetworkSecurityGroupRuleAdd(ctx context.Context, client *instance.IBMPINetworkSecurityGroupClient, id, ruleID string, timeout time.Duration) (interface{}, error) {
+
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{State_Pending},
+		Target:     []string{State_Available},
+		Refresh:    isIBMPINetworkSecurityGroupRuleAddRefreshFunc(client, id, ruleID),
+		Timeout:    timeout,
+		Delay:      10 * time.Second,
+		MinTimeout: 10 * time.Minute,
```

Comment:
```suggestion
		MinTimeout: 10 * time.Minute,
```
This can be reduce to 60 sec

Reply:
updated